File: src/main/java/org/polyfrost/glintcolorizer/GlintColorizer.java
package org.polyfrost.glintcolorizer;

import com.mojang.brigadier.Command;
import org.polyfrost.glintcolorizer.config.GlintColorizerConfig;

import net.fabricmc.api.ClientModInitializer;
import org.polyfrost.oneconfig.api.commands.v1.CommandManager;
import org.polyfrost.oneconfig.utils.v1.dsl.ScreensKt;

public final class GlintColorizer implements ClientModInitializer {
	@Override
	public void onInitializeClient() {
		// Config
		GlintColorizerConfig.INSTANCE.preload();

		// Commands
        CommandManager.INSTANCE.register(
            CommandManager.literal("glintcolorizer")
            .executes((source) -> {
                ScreensKt.openUI(GlintColorizerConfig.INSTANCE);
                return Command.SINGLE_SUCCESS;
            })
        );
    }
}

File: src/main/kotlin/org/polyfrost/glintcolorizer/GlintColorizer.kt
package org.polyfrost.glintcolorizer


import net.fabricmc.api.ClientModInitializer
import net.ornithemc.osl.lifecycle.api.client.MinecraftClientEvents

import org.polyfrost.glintcolorizer.command.GlintCommand
import org.polyfrost.glintcolorizer.config.GlintConfig
import org.polyfrost.oneconfig.api.commands.v1.CommandManager

object GlintColorizer : ClientModInitializer {
    const val NAME: String = "@MOD_NAME@"
    const val VERSION: String = "@MOD_VERSION@"
    const val ID: String = "@MOD_ID@"

    override fun onInitializeClient() {
        CommandManager.register(GlintCommand())
        MinecraftClientEvents.READY.register() { context: Any -> GlintConfig.preload() }
    }
}
